ReactApp/src/misc/api-fetch-setup.js

```diff
@@ -3,26 +3,24 @@
  */
 import apiFetch from '@wordpress/api-fetch';
 
+/**
+ * Internal dependencies
+ */
+import { getGBKit } from './store';
+
 /**
  * Initializes the API fetch configuration and middleware.
  *
  * This function sets up the root URL middleware, adds headers to requests,
  * and preloads some endpoints with mock data for specific components.
  */
 export function initializeApiFetch() {
-	apiFetch.use(apiFetch.createRootURLMiddleware(''));
+	const { siteURL, authToken } = getGBKit();
+	const rootURL = siteURL ? `${siteURL}/wp-json/` : undefined;
```

Comment:
Interesting. I make successful requests with this approach for an Atomic WPCOM site. Is the use of `https://public-api.wordpress.com/` merely a recommendation and/or potentially required for Simple sites?

Reply:
I was trying with a simple site yeah. For Atomic, it's the same as self-hosted using the site's URL, but for Simple sites I think we'll need to go through the WordPress.com API 🤔 

I ended up trying this because I checked the editor's Web view source code and for Simple sites, it sets `https://public-api.wordpress.com/` for the `createRootURLMiddleware` middleware and for Atomic, it just uses the site's URL e.g. `https://geriuxxtestingbusiness123.wpcomstaging.com/wp-json/`

Reply:
Makes sense. When we address this, we might consider renaming `siteURL` to something like `siteAPIURL`.

Reply:
Addressed [here](https://github.com/wordpress-mobile/GutenbergKit/pull/15/files#diff-230d4e17b4d9726d839cbab42a1cb29f2ad91e6ada136a558c6df6e421b4976fR18)